llvmlite/binding/value.py

```diff
@@ -299,6 +344,43 @@ def opcode(self):
                              % (self._kind,))
         return ffi.ret_string(ffi.lib.LLVMPY_GetOpcodeName(self))
 
+    def get_constant_value(self):
+        """
+        Return the constant value, either as a literal (when supported)
+        or as a string.
+        """
+        if not self.is_constant:
+            raise ValueError('expected constant value, got %s'
+                             % (self._kind,))
+
+        if self.value_kind == ValueKind.constant_int:
+            # Python integers are also arbitrary-precision
+            little_endian = c_bool(False)
+            words = ffi.lib.LLVMPY_GetConstantIntNumWords(self)
+            ptr = ffi.lib.LLVMPY_GetConstantIntRawValue(
+                self, byref(little_endian))
+            asbytes = bytes(cast(ptr, POINTER(c_uint64 * words)).contents)
+            return int.from_bytes(
+                asbytes,
+                ('little' if little_endian.value else 'big'),
+                signed=False,
+            )
+        elif self.value_kind == ValueKind.constant_fp:
+            # Convert floating-point values to double (Python float)
+            accuracy_loss = c_bool(False)
+            value = ffi.lib.LLVMPY_GetConstantFPValue(self,
+                                                      byref(accuracy_loss))
+            if accuracy_loss.value:
+                warnings.warn(
```

Comment:
Needs to add test for the warning

Reply:
@sklam For some reason, despite how much I try, I can't seem to trigger the `losesInfo` boolean with a single constant (maybe this appears during passes). Not sure how to proceed here, any suggestions?

Reply:
Sorry for the delay. 

After some digging into the LLVM source, the floating-point constants are encoded as double-precision so there will not be any information loss here. To force info loss, we will need to use bigger types like the long-double. For example: `%const = fadd fp128 0xLF3CB1CCF26FBC178452FB4EC7F91DEAD, 0xLF3CB1CCF26FBC178452FB4EC7F91973F`. This will require converting the 128-bit float to 64-bit float thus triggering the warning. 

However, this brings to my attention that a warning is not appropriate here. This API is not giving user a choice to read float bigger than double-precision.  In the LLVM OCAML binding, the similar function will return `None` (https://github.com/llvm/llvm-project/blob/1defa781243f9d0bc66719465e4de33e9fb7a243/llvm/bindings/ocaml/llvm/llvm_ocaml.c#L1030-L1032). I think the Python API should raise an exception.

Also, maybe it is better to have separate functions for int constant and float constant, so that the float version can provide more control on what to do if information is lost. For instance, if user is okay with the info loss, they can force the downcast.

Reply:
That's great, I agree and already modified the constant float test. Only thing I'm not sure about is the separate functions for int and float constants. One of the main benefits of Python is that you can return multiple types. Maybe this should be another keyword argument, and keyword args should act like preferences?

I am already extending the behavior to support other types of constants (e.g. structs), and it was beneficial to be able to call the same method on the nested data: https://github.com/tbennun/llvmlite/commit/3992ea175c164c9deedb8644b12b138c2e0c0dfc#diff-27ab06b9fa73e854c37e5ccf261269588cd8e02fadbd71c788102bc46c4a2f1bR390

Reply:
@sklam all changes are complete and the PR is ready for re(re)review.